src/main/java/dev/esophose/playerparticles/particles/ParticleEffect.java

```diff
@@ -378,8 +378,8 @@ public static void display(ParticlePair particle, PParticle pparticle, boolean i
             effect.display(data, pparticle.getXOff(), pparticle.getYOff(), pparticle.getZOff(), 1, pparticle.getLocation(false), isLongRange, owner);
         } else if (effect.hasProperty(ParticleProperty.VIBRATION)) {
             Vibration data;
-            if (pparticle.getOverrideData() instanceof Vibration) {
-                data = (Vibration) pparticle.getOverrideData();
+            if (pparticle.getOverrideData() instanceof org.bukkit.Vibration) {
```

Comment:
Please keep this using the PlayerParticles version of the Vibration object. With the other changes you've made this shouldn't be a problem.